### ISSUE TITLE
host field in _cat/nodes is always an IP address

### ISSUE THREAD

**Issue by sysmonk** [elastic/elasticsearch#16575]
I've upgraded my elasticsearch cluster from 1.7.4 to 2.2.0 yesterday, and i noticed that _cat/nodes?v now always returns an IP address in the "host" field.

ElasticSearch 2.2 Example:
```
host          ip            heap.percent ram.percent load node.role master name
192.168.0.75   192.168.0.75            68          99 1.07 d         m      es1.logstash.example.com     
192.168.0.55   192.168.0.55            64         100 0.44 d         m      es2.logstash.example.com      
192.168.0.69   192.168.0.69            29          98 1.09 d         m      es4.logstash.example.com     
192.168.0.79   192.168.0.79            30         100 1.14 d         m      es6.logstash.example.com     
192.168.0.246  192.168.0.246           68         100 1.19 d         m      es5.logstash.example.com      
192.168.0.49   192.168.0.49            73          97 0.62 d         *      es8.logstash.example.com      

```

Previously, in ElasticSearch 1.7.4 it had the 'host' field populated with the hostname (same as the 'name' field). 
I've tried setting network.bind_host network.publish_host network.host to the hostnames, even tried adding PTR record in case ES does PTR lookups, but the host field is always an IP address.

Other, kinda related issues, is that the _cat/nodes list is sorted randomly at every refresh. In 1.7.4 it was also "random" - except that the randomness didn't change with every refresh, but only when nodes leave/join cluster.

**Comment by dakrone** [comment]
I was able to reproduce this and see yeah, the hosts are ip addresses instead of hostnames.

> Other, kinda related issues, is that the _cat/nodes list is sorted randomly at every refresh.

Probably makes sense to sort the nodes so they appear in a consistent order for the output

**Comment by jasontedor** [comment]
It was an [intentional choice](https://github.com/elastic/elasticsearch/pull/12959) to remove hostname resolving for the underlying `TransportAddress`es. This leads to the `host` and `ip` fields always being equal here. As such, the `host` field here should probably just be removed from the cat nodes API; if you need a name one appears as the `name` field in the cat nodes API (already shown in your output); this is something that you can completely control via node configuration.

As for the sorting, the design of cat API is meant to be friendly with standard command line tools. In this case, if you need to sort on `name` you can just do:

```
$ curl -XGET locahost:9200/_cat/nodes | sort -k8
```

(until the default output changes to have the `host` field removed).

**Comment by jasontedor** [comment]
Closed by #16656
